packages/dataviews/src/view-list.js

```diff
@@ -26,9 +144,15 @@ export default function ViewList( {
 	onDetailsChange,
 	selection,
 	deferredRendering,
+	id: preferredId,
 } ) {
+	const baseId = useInstanceId( ViewList, 'view-list', preferredId );
 	const shownData = useAsyncList( data, { step: 3 } );
 	const usedData = deferredRendering ? shownData : data;
+	const selectedItem = usedData?.findLast( ( item ) =>
```

Comment:
This changes how selection works for the list item. Particularly, in the case of some items being pre-selected from a different view and then switch to the list view.

Before:

https://github.com/WordPress/gutenberg/assets/583546/6fdec80f-272a-4fb5-a4af-333c5d8fa4f4

After:

https://github.com/WordPress/gutenberg/assets/583546/5cab223e-79b2-406a-9897-c2ae494b2a8e

Until we figure out whether it makes sense to have multiple selected elements in the list view, this is a good approach.